external/storm-hbase/src/main/java/org/apache/storm/hbase/bolt/HBaseLookupBolt.java

```diff
@@ -40,51 +48,81 @@
  *
  */
 public class HBaseLookupBolt extends AbstractHBaseBolt {
-    private static final Logger LOG = LoggerFactory.getLogger(HBaseLookupBolt.class);
+	
+   private static final long serialVersionUID = 1L;
+   private static final Logger LOG = LoggerFactory.getLogger(HBaseLookupBolt.class);
+
+   private HBaseValueMapper rowToTupleMapper;
+   private HBaseProjectionCriteria projectionCriteria;
+   private transient LoadingCache<byte[], Result> cache;
+   private transient boolean cacheEnabled;
 
-    private HBaseValueMapper rowToTupleMapper;
+   public HBaseLookupBolt(String tableName, HBaseMapper mapper, HBaseValueMapper rowToTupleMapper) {
+      super(tableName, mapper);
+      Validate.notNull(rowToTupleMapper, "rowToTupleMapper can not be null");
+      this.rowToTupleMapper = rowToTupleMapper;
+   }
 
-    private HBaseProjectionCriteria projectionCriteria;
+   public HBaseLookupBolt withConfigKey(String configKey) {
+      this.configKey = configKey;
+      return this;
+   }
 
-    public HBaseLookupBolt(String tableName, HBaseMapper mapper, HBaseValueMapper rowToTupleMapper){
-        super(tableName, mapper);
-        Validate.notNull(rowToTupleMapper, "rowToTupleMapper can not be null");
-        this.rowToTupleMapper = rowToTupleMapper;
-    }
+   public HBaseLookupBolt withProjectionCriteria(HBaseProjectionCriteria projectionCriteria) {
+      this.projectionCriteria = projectionCriteria;
+      return this;
+   }
 
-    public HBaseLookupBolt withConfigKey(String configKey){
-        this.configKey = configKey;
-        return this;
-    }
+   @SuppressWarnings({ "unchecked", "rawtypes" })
+   @Override
+   public void prepare(Map config, TopologyContext topologyContext, OutputCollector collector) {
+      super.prepare(config, topologyContext, collector);
+      cacheEnabled = Boolean.parseBoolean(config.getOrDefault("hbase.cache.enable", "false").toString());
+      int cacheTTL = Integer.parseInt(config.getOrDefault("hbase.cache.ttl.seconds", "300").toString());
+      int maxCacheSize = Integer.parseInt(config.getOrDefault("hbase.cache.size", "1000").toString());
+      if (cacheEnabled) {
+         cache = Caffeine.newBuilder()
+        		    .maximumSize(maxCacheSize).expireAfterWrite(cacheTTL, TimeUnit.SECONDS)
+               .build(new CacheLoader<byte[], Result>() {
 
-    public HBaseLookupBolt withProjectionCriteria(HBaseProjectionCriteria projectionCriteria) {
-        this.projectionCriteria = projectionCriteria;
-        return this;
-    }
+                  @Override
+                  public Result load(byte[] rowKey) throws Exception {
+                     Get get = hBaseClient.constructGetRequests(rowKey, projectionCriteria);
+                     LOG.debug("Cache miss for key:"+new String(rowKey));
```

Comment:
@ambud
It would be better to use `Cache miss for key: {}` instead of `+`. After fixing this, you can squash commits into one.